Source/Core/Core/AchievementManager.h

```diff
@@ -60,17 +70,29 @@ class AchievementManager
   ResponseType FetchUnlockData(bool hardcore);
 
   void ActivateDeactivateAchievement(AchievementId id, bool enabled, bool unofficial, bool encore);
+  RichPresence GenerateRichPresence();
+
+  ResponseType AwardAchievement(AchievementId achievement_id);
+  ResponseType SubmitLeaderboard(AchievementId leaderboard_id, int value);
+  ResponseType PingRichPresence(RichPresence rich_presence);
+
+  void HandleAchievementTriggeredEvent(const rc_runtime_event_t* runtime_event);
+  void HandleLeaderboardTriggeredEvent(const rc_runtime_event_t* runtime_event);
 
   template <typename RcRequest, typename RcResponse>
   ResponseType Request(RcRequest rc_request, RcResponse* rc_response,
                        const std::function<int(rc_api_request_t*, const RcRequest*)>& init_request,
                        const std::function<int(RcResponse*, const char*)>& process_response);
 
   rc_runtime_t m_runtime{};
+  const Core::System& m_system = Core::System::GetInstance();
```

Comment:
This should be assigned as a non-const reference. Also please do this by placing the the constructor in the cpp file like:

```cpp
AchievementManager::AchievementManager(Core::System& system)
  : m_system{Core::System::GetInstance()} {}
```

and remove

```cpp
#include "Core/Core.h"
#include "Core/System.h"
```

from the header and move them to the cpp file, and in their place, provide the forward declaration:

```cpp
namespace Core
{
class System;
}
```

There's no need to dump such large includes into the header. Or just use the accessor directly inside the implementation. This really doesn't need to be complicated for such a small part of the codebase, which is still going through refactoring and implementation work.